Gems/ScriptAutomation/Code/Include/ScriptAutomation/ScriptAutomationBus.h

```diff
@@ -61,8 +65,14 @@ namespace ScriptAutomation
         //! tell the automation system that a profiling capture has started
         virtual void StartProfilingCapture() = 0;
 
+        //! Call out to another script and put it in the operation queue
+        virtual void ExecuteScript(const char* scriptPath) = 0;
+
         //! Add an operation into the queue for processing later
         virtual void QueueScriptOperation(ScriptOperation&& action) = 0;
+
+        //! Find the named tolerance level
+        virtual const ImageComparisonToleranceLevel* FindToleranceLevel(const AZStd::string& name) = 0;
```

Comment:
A forward declared type can be returned by value, as long as the function is definition is defined when the ImageComparisonToleranceLevel class is available: https://godbolt.org/z/GqGo3WdEf

Reply:
Ok that makes sense if the value is optional.

Reply:
Confirmed that the current implementation will return a nullptr if the requested level isn't found.

In this case optional seems like overkill, there is only 1 error case rather than many where the ability to distinguish the errors would be useful.